In [1]:
import pandas as pd
import numpy as np
import os

data_path = r"C:\Users\likhi\OneDrive\Desktop\e commerce project\data\\"

print("Files found:")
for f in os.listdir(data_path):
    print(" ", f)

Files found:
  olist_customers_dataset.csv
  olist_geolocation_dataset.csv
  olist_orders_dataset.csv
  olist_order_items_dataset.csv
  olist_order_payments_dataset.csv
  olist_order_reviews_dataset.csv
  olist_products_dataset.csv
  olist_sellers_dataset.csv
  product_category_name_translation.csv


In [2]:
orders       = pd.read_csv(data_path + "olist_orders_dataset.csv", encoding='utf-8', low_memory=False)
order_items  = pd.read_csv(data_path + "olist_order_items_dataset.csv", encoding='utf-8', low_memory=False)
customers    = pd.read_csv(data_path + "olist_customers_dataset.csv", encoding='utf-8', low_memory=False)
products     = pd.read_csv(data_path + "olist_products_dataset.csv", encoding='utf-8', low_memory=False)
sellers      = pd.read_csv(data_path + "olist_sellers_dataset.csv", encoding='utf-8', low_memory=False)
reviews      = pd.read_csv(data_path + "olist_order_reviews_dataset.csv", encoding='utf-8', low_memory=False)
payments     = pd.read_csv(data_path + "olist_order_payments_dataset.csv", encoding='utf-8', low_memory=False)
geolocation  = pd.read_csv(data_path + "olist_geolocation_dataset.csv", encoding='utf-8', low_memory=False)
category_translation = pd.read_csv(data_path + "product_category_name_translation.csv", encoding='utf-8', low_memory=False)

print("All 9 tables loaded successfully")

All 9 tables loaded successfully


In [3]:
def inspect_df(name, df):
    print(f"\n{'='*50}")
    print(f"TABLE: {name}")
    print(f"Rows: {df.shape[0]}  |  Columns: {df.shape[1]}")
    print(f"\nNull counts:")
    print(df.isnull().sum())
    print(f"\nSample:")
    display(df.head(3))

inspect_df("orders", orders)
inspect_df("order_items", order_items)
inspect_df("customers", customers)
inspect_df("products", products)
inspect_df("sellers", sellers)
inspect_df("reviews", reviews)
inspect_df("payments", payments)
inspect_df("category_translation", category_translation)


TABLE: orders
Rows: 99441  |  Columns: 8

Null counts:
order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
order_estimated_delivery_date       0
dtype: int64

Sample:


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00



TABLE: order_items
Rows: 112650  |  Columns: 7

Null counts:
order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64

Sample:


,order_id,order_item_id,product_id,seller_id,shipping_limit_date,price,freight_value
0,00010242fe8c5a6d1ba2dd792cb16214,1,4244733e06e7ecb4970a6e2683c13e61,48436dade18ac8b2bce089ec2a041202,2017-09-19 09:45:35,58.9,13.29
1,00018f77f2f0320c557190d7a144bdd3,1,e5f2d52b802189ee658865ca93d83a8f,dd7ddc04e1b6c2c614352b383efe2d36,2017-05-03 11:05:13,239.9,19.93
2,000229ec398224ef6ca0657da4fc703e,1,c777355d18b72b67abbeef9df44fd0fd,5b51032eddd242adc84c38acab88f23d,2018-01-18 14:48:30,199.0,17.87



TABLE: customers
Rows: 99441  |  Columns: 5

Null counts:
customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64

Sample:


,customer_id,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state
0,06b8999e2fba1a1fbc88172c00ba8bc7,861eff4711a542e4b93843c6dd7febb0,14409,franca,SP
1,18955e83d337fd6b2def6b18a428ac77,290c77bc529b7ac935b93aa66c333dc3,9790,sao bernardo do campo,SP
2,4e7b3e00288586ebd08712fdd0374a03,060e732b5b29e8181a18229c7b0b2b5e,1151,sao paulo,SP



TABLE: products
Rows: 32951  |  Columns: 9

Null counts:
product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product_length_cm               2
product_height_cm               2
product_width_cm                2
dtype: int64

Sample:


,product_id,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm
0,1e9e8ef04dbcff4541ed26657ea517e5,perfumaria,40.0,287.0,1.0,225.0,16.0,10.0,14.0
1,3aa071139cb16b67ca9e5dea641aaa2f,artes,44.0,276.0,1.0,1000.0,30.0,18.0,20.0
2,96bd76ec8810374ed1b65e291975717f,esporte_lazer,46.0,250.0,1.0,154.0,18.0,9.0,15.0



TABLE: sellers
Rows: 3095  |  Columns: 4

Null counts:
seller_id                 0
seller_zip_code_prefix    0
seller_city               0
seller_state              0
dtype: int64

Sample:


,seller_id,seller_zip_code_prefix,seller_city,seller_state
0,3442f8959a84dea7ee197c632cb2df15,13023,campinas,SP
1,d1b65fc7debc3361ea86b5f14c68d2e2,13844,mogi guacu,SP
2,ce3ad9de960102d0677a81f5d0bb7b2d,20031,rio de janeiro,RJ



TABLE: reviews
Rows: 99224  |  Columns: 7

Null counts:
review_id                      0
order_id                       0
review_score                   0
review_comment_title       87656
review_comment_message     58247
review_creation_date           0
review_answer_timestamp        0
dtype: int64

Sample:


,review_id,order_id,review_score,review_comment_title,review_comment_message,review_creation_date,review_answer_timestamp
0,7bc2406110b926393aa56f80a40eba40,73fc7af87114b39712e6da79b0a377eb,4,NaN,NaN,2018-01-18 00:00:00,2018-01-18 21:46:59
1,80e641a11e56f04c1ad469d5645fdfde,a548910a1c6147796b98fdf73dbeba33,5,NaN,NaN,2018-03-10 00:00:00,2018-03-11 03:05:13
2,228ce5500dc1d8e020d8d1322874b6f0,f9e4b658b201a9f2ecdecbb34bed034b,5,NaN,NaN,2018-02-17 00:00:00,2018-02-18 14:36:24



TABLE: payments
Rows: 103886  |  Columns: 5

Null counts:
order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64

Sample:


,order_id,payment_sequential,payment_type,payment_installments,payment_value
0,b81ef226f3fe1789b1e8b2acac839d17,1,credit_card,8,99.33
1,a9810da82917af2d9aefd1278f1dcfa0,1,credit_card,1,24.39
2,25e8ea4e93396b6fa0d3dd708e76c1bd,1,credit_card,1,65.71



TABLE: category_translation
Rows: 71  |  Columns: 2

Null counts:
product_category_name            0
product_category_name_english    0
dtype: int64

Sample:


,product_category_name,product_category_name_english
0,beleza_saude,health_beauty
1,informatica_acessorios,computers_accessories
2,automotivo,auto


In [4]:
print("Orders unique order_ids:", orders['order_id'].nunique())
print("Items unique order_ids:", order_items['order_id'].nunique())
print("Reviews unique order_ids:", reviews['order_id'].nunique())

orphaned = order_items[~order_items['order_id'].isin(orders['order_id'])]
print(f"\nOrphaned item rows (no matching order): {len(orphaned)}")

Orders unique order_ids: 99441
Items unique order_ids: 98666
Reviews unique order_ids: 98673

Orphaned item rows (no matching order): 0


## Data Inspection Observations

### orders (99,441 rows)
- 160 nulls in order_approved_at — likely cancelled orders
- 2,965 nulls in order_delivered_customer_date — orders never delivered
- All date columns are strings — will convert to datetime in cleaning

### products (32,951 rows)
- 610 nulls in product_category_name — products with no category
- Will join with category_translation for English names

### reviews (99,224 rows)
- review_score has zero nulls — key column for analysis
- 88% have no comment title, 59% no comment message — normal

### order_items, customers, payments, sellers
- All completely clean, zero nulls

### Key things to fix in cleaning phase
- Parse date columns to datetime
- Engineer delivery_days, late_delivery, order_month columns
- Join all tables into one master dataframe
- Export clean_orders.csv to outputs folder